mapbox_vector_tile/encoder.py

```diff
@@ -56,13 +56,28 @@ def transform(shape, func):
     raise ValueError('Unknown geometry type, "%s"' % shape.type)
 
 
+def on_invalid_geometry_raise(shape):
+    raise Exception('Invalid geometry: %s' % shape.wkt)
+
+
+def on_invalid_geometry_ignore(shape):
+    return None
+
+
+def on_invalid_geometry_make_valid(shape):
+    if shape.type in ('Polygon', 'MultiPolygon'):
+        shape = shape.buffer(0)
```

Comment:
Do we want to `assert shape.is_valid` after this, just to make sure that we are actually making the shape valid. Alternatively, we could chain some of these together, e.g: `make_valid.andThen(ignore)` or something like that?